Makefile

```diff
@@ -80,21 +80,21 @@ docker-build:
 	docker build -t pydo:dev .
 
 .PHONY: docker-python
-docker-python: docker-build  ## Runs a python shel within a docker container
+docker-python: docker-build  ## Runs a python shell within a docker container
 	docker run -it --rm --name pydo pydo:dev python
 
 .PHONY: lint-docs
 lint-docs:
 	docker run -v $(ROOT_DIR):/workdir ghcr.io/igorshubovych/markdownlint-cli:latest "*.md"
 
 .PHONY: generate-docs
-generate-docs: install ## Generate documentation for Client using Sphinx
+generate-docs: install ## readthedocs requires a requirements.txt file, this step converts poetry file to requirements.txt file before re-gen the docs
 	@echo Generating documentation...;
 	@echo Converting poetry file to requirements.txt...; \
 	poetry export -f requirements.txt -o requirements.txt --without-hashes && \
 	cd docs && \
-	sphinx-apidoc -o source/ ../src/pydo && \
-	make html
+	poetry run sphinx-apidoc -o source/ ../src/pydo && \
+	poetry run make html
 
 .PHONY: clean-docs
 clean-docs: ## Delete everything in docs/build/html
```

Comment:
I know this isn't part of this change but I also noticed this command fails because it's run outside of poetry's virtual env. Changing:

```
	make clean
```

To this should fix it:
```
	poetry run make clean
```

Reply:
good catch!